qtfred/src/mission/management.cpp

```diff
@@ -26,75 +35,220 @@ extern void allocate_mission_text(size_t size);
 
 extern void parse_init(bool basic = false);
 
+extern void brief_init_colors();
+
+extern void ssm_init();    // Need this to populate Ssm_info so OPF_SSM_CLASS does something. -MageKing17
+
+extern void cmdline_debug_print_cmdline();
+
 namespace fso {
 namespace fred {
 
+void fred_preload_all_briefing_icons() {
+	for (SCP_vector<briefing_icon_info>::iterator ii = Briefing_icon_info.begin(); ii != Briefing_icon_info.end();
+		 ++ii) {
+		generic_anim_load(&ii->regular);
+		hud_anim_load(&ii->fade);
+		hud_anim_load(&ii->highlight);
+	}
+}
+
+bool
+initialize(const std::string& cfilepath, int argc, char* argv[], Editor* editor, const InitializerCallback& listener) {
+	std::setlocale(LC_ALL, "C");
+
+	srand((unsigned) time(NULL));
 
-void initialize(const std::string& cfilepath, int argc, char* argv[], Editor* editor, InitializerCallback listener) {
+	listener(SubSystem::OS);
+	os_init(Osreg_class_name, Osreg_app_name);
+
+	listener(SubSystem::Timer);
+	timer_init();
+
+	listener(SubSystem::CommandLine);
+	// this should enable mods - Kazan
+	parse_cmdline(argc, argv);
 
 #ifndef NDEBUG
-	outwnd_init();
+#if FS_VERSION_REVIS == 0
+	mprintf(("qtFred Open version: %i.%i.%i\n", FS_VERSION_MAJOR, FS_VERSION_MINOR, FS_VERSION_BUILD));
+#else
+	mprintf(("qtFred Open version: %i.%i.%i.%i\n", FS_VERSION_MAJOR, FS_VERSION_MINOR, FS_VERSION_BUILD, FS_VERSION_REVIS));
+#endif
+
+	cmdline_debug_print_cmdline();
+#endif
+
+	// d'oh
+	// cfile needs a file path...
+	auto file_path = cfilepath + DIR_SEPARATOR_STR + "qtfred.exe";
+
+	listener(SubSystem::CFile);
+	if (cfile_init(file_path.c_str())) {
+		return false;
+	}
 
+	listener(SubSystem::ModTable);
+	// Load game_settings.tbl
+	mod_table_init();
+
+	listener(SubSystem::Locale);
+	// initialize localization module. Make sure this is done AFTER initialzing OS.
+	// NOTE : Fred should ALWAYS run in English. Otherwise it might swap in another language
+	// when saving - which would cause inconsistencies when externalizing to tstrings.tbl via Exstr
+	// trust me on this :)
+	lcl_init(FS2_OPEN_DEFAULT_LANGUAGE);
+
+	// Goober5000 - force init XSTRs (so they work, but only work in English, based on above comment)
+	Xstr_inited = 1;
+
+#ifndef NDEBUG
 	load_filter_info();
 #endif
 
-	std::setlocale(LC_ALL, "C");
+	//CFREDView *window = CFREDView::GetView();
+	//HWND hwndApp = window->GetSafeHwnd();
+	//os_set_window((uint) hwndApp);
```

Comment:
Can you remove this commented out code if we don't need it?

Reply:
Fixed.